src/ragger/navigator/navigator.py

```diff
@@ -437,42 +531,44 @@ def navigate_until_text_and_compare(self,
         :rtype: NoneType
         """
         idx = 0
-        snaps_tmp_path = None
-        snaps_golden_path = None
-
-        if path and test_case_name:
-            snaps_tmp_path = self._init_snaps_temp_dir(path, test_case_name)
-            snaps_golden_path = self._check_snaps_dir_path(path, test_case_name, True)
-
+        start = time()
         if not isinstance(self._backend, SpeculosBackend):
             # TODO remove this once the proper behavior of other backends is implemented.
             return
 
-        start = time()
-
-        if screen_change_before_first_instruction:
-            self._backend.wait_for_screen_change(timeout)
+        # Navigation initialization: no-op instruction to:
+        # - wait for screen change depending on screen_change_before_first_instruction.
+        #   this is necessary when an APDU was just send and we want to make sure the
+        #   screen already display the first review page.
+        # - compare the initial screen content with the golden reference if path and
+        #   test_case_name are valid.
+        self._run_instruction(NavIns(NavInsID.WAIT, (0, )),
+                              timeout,
+                              wait_for_screen_change=screen_change_before_first_instruction,
+                              path=path,
+                              test_case_name=test_case_name,
+                              snap_idx=idx)
```

Comment:
This feels a bit like a hack as we are not interested in 90% of what the `_run_instruction` does but I don't really have better to offer

Reply:
Humf,
`_run_instruction()` does:

1. retrieve and execute the isntruction callback
2. wait for screen change
3. compare screen snapshot if requested.

So basically we want all of this.
But as it's a simple instruction, the wait for change is simpler and thta's what takes a lot of space in `_run_instruction()`

Reply:
Ok